plugins/outputs/cloudwatch/cloudwatch.go

```diff
@@ -279,48 +276,90 @@ func (c *CloudWatch) timeToPublish(b *MetricDatumBatch) bool {
 	return len(b.Partition) > 0 && time.Now().Sub(b.BeginTime) >= c.ForceFlushInterval.Duration
 }
 
+// getFirstPushMs returns the time at which the first upload should occur.
+// It uses random jitter as an offset from the start of the given interval.
+func getFirstPushMs(interval time.Duration) int64 {
+	publishJitter := publishJitter(interval)
+	log.Printf("I! cloudwatch: publish with ForceFlushInterval: %v, Publish Jitter: %v",
+		interval, publishJitter)
+	nowMs := time.Now().UnixMilli()
+	// Truncate i.e. round down, then add jitter.
+	// If the rounded down time is in the past, move it forward.
+	nextMs := nowMs - (nowMs % interval.Milliseconds()) + publishJitter.Milliseconds()
+	if nextMs < nowMs {
+		nextMs += interval.Milliseconds()
+	}
+	return nextMs
+}
+
+// publish will begin by sleeping for a random amount of jitter between 0 and
+// the configured forceFlushInterval. Previously, if the batch buffer filled up
+// it flushed immediately. Now it cuts the remaining interval in half,
+// and the next interval in half.
+// If the buffer continues to stay full then it will keep cutting down.
+// This avoids bursting the backend and maintains some amount of jitter.
 func (c *CloudWatch) publish() {
-	now := time.Now()
-	forceFlushInterval := c.ForceFlushInterval.Duration
-	publishJitter := publishJitter(forceFlushInterval)
-	log.Printf("I! cloudwatch: publish with ForceFlushInterval: %v, Publish Jitter: %v", forceFlushInterval, publishJitter)
-	time.Sleep(now.Truncate(forceFlushInterval).Add(publishJitter).Sub(now))
```

Comment:
Just a note, this old call to `time.Sleep()` could be called with a negative value.
If current time was `04:36:47` and the flush interval was `60s`, then it rounds down to `04:36:00` then add jitter.
If the jitter was was anything less than `47s` it wouldn't sleep at all.
So then the jitter is only based on when the agent process started.